### PR TITLE
add dev-env

### PR DESCRIPTION
**Pull request by garyverhaegen-da**
I got mildly annoyed at having to #259 again, and then @cocreature pushed me over the edge, so here is a proper fix.

Note: this only include Terraform in the dev-env for now, but we should probably look at expanding that over time.

This also means Terraform gets bumped again, to 0.12.28, because I was too lazy to hunt down the latest nixpkgs snapshot that still included 0.12.26, so instead I just picked the latest one.

Also note that nix-bundled Terraform comes with its embedded plugins, which do not have a version string, so we need to remove the version requirement from the tf file.

### CODE REVIEW

File: dev-env/lib/dade-base-profile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# DEPRECATED. Base profile for both bash and zsh -*- shell-script -*-
+#
+# TODO(gleber): delete when `da-ops` and `nixBash` switches to `dade-dump-profile`.
+# Sourcable wrapper around `dade-dump-profile`. See it for details.
+
+# TODO(gleber): Compatibility mode until JBH is ugraded.
```
Comment (cocreature) [inline dev-env/lib/dade-base-profile:6]: I don’t even know what JBH stands for. I wonder if we should just kill thaf file everywhere. I don’t know of anyone using it in the last few months.

File: dev-env/lib/dade-dump-profile [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+    # since we need to set LC_ALL=C to prevent any messages about missing locale
+    # until we fetch the locale, we start in a separate shell
+    (
+        export LC_ALL=C
+        unset NIX_PATH
+        out=$(nix-build --no-out-link -Q "${DADE_DEVENV_DIR}/../nix" -I "${TMP_NIX_PATH}" -A pkgs.glibcLocales)
+        echo "export LOCALE_ARCHIVE=\"${out}/lib/locale/locale-archive\""
+    )
+fi
+echo "export LC_ALL=en_US.UTF-8"
```
Comment (cocreature) [inline dev-env/lib/dade-dump-profile:128]: Same thing here, I don’t know of anyone using this.

File: dev-env/lib/dade-base-profile (hunk 1 above)
Comment (garyverhaegen-da) [reply dev-env/lib/dade-base-profile:6]: This file is not actually mentioned anywhere else, so I suppose it's unused. Deleting from this repo at least, I'll double-check on daml. 🤷

File: dev-env/lib/dade-dump-profile (hunk 2 above)
Comment (garyverhaegen-da) [reply dev-env/lib/dade-dump-profile:128]: `dade-assist` is basically just calling this one, and I'm pretty sure we use `dade-assist` in a few places :-)

Comment (cocreature) [reply dev-env/lib/dade-dump-profile:128]: Oh makes sense, `dump-profile` sounded like it was some debugging thing. Should have checked

File: dev-env/lib/dade-base-profile (hunk 1 above)
Comment (garyverhaegen-da) [reply dev-env/lib/dade-base-profile]: This has been bothering me for the past two hours, so now that I just remembered it, I need to share: JBH stands for `Jenkins Build Helpers`. That was the name for the library of Groovy(-ish; "Jenkins declarative pipeline" scripts) code the BRE team was maintaining to make creating builds on the old Jenkins instance easier.

Comment (cocreature) [reply dev-env/lib/dade-base-profile]: Nice, thanks for doing some archeology :)

**Pull request merged**